core/domain/platform_parameter_domain_test.py

```diff
@@ -81,3 +82,1099 @@ def test_to_dict(self):
+            'data_type': 'InvalidType',
+            'rules': [
+                {
+                    'filters': [
+                        {
+                            'type': 'server_mode',
+                            'conditions': [('=', 'dev')]
+                        }
+                    ],
+                    'value_when_matched': '222'
+                }
+            ],
+            'rule_schema_version': (
+                feconf.CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION),
+            'default_value': '333',
+            'metadata': {
+                'is_feature': False,
+                'feature_stage': None,
+            },
+        })
+        with self.assertRaisesRegexp(
+            utils.ValidationError, 'Unsupported data type \'InvalidType\''):
+            param.validate()
+
+    def test_validate_with_inconsistent_data_type_in_rules_raises_exception(
+            self):
+        param = parameter_domain.PlatformParameter.from_dict({
+            'name': 'parameter_a',
+            'description': 'for test',
+            'data_type': 'bool',
+            'rules': [
+                {
+                    'filters': [
+                        {
+                            'type': 'server_mode',
+                            'conditions': [('=', 'dev')]
+                        }
+                    ],
+                    'value_when_matched': '222'
+                },
+            ],
+            'rule_schema_version': (
+                feconf.CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION),
+            'default_value': False,
+            'metadata': {
+                'is_feature': False,
+                'feature_stage': None,
+            },
+        })
+        with self.assertRaisesRegexp(
+            utils.ValidationError,
+            'Expected bool, received \'222\' in value_when_matched'):
+            param.validate()
+
+    def test_validate_with_inconsistent_default_value_type_raises_exception(
+            self):
+        param = parameter_domain.PlatformParameter.from_dict({
+            'name': 'parameter_a',
+            'description': 'for test',
+            'data_type': 'bool',
+            'rules': [],
+            'rule_schema_version': (
+                feconf.CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION),
+            'default_value': '111',
+            'metadata': {
+                'is_feature': False,
+                'feature_stage': None,
+            },
+        })
+        with self.assertRaisesRegexp(
+            utils.ValidationError,
+            'Expected bool, received \'111\' in default value'):
+            param.validate()
+
+    def test_create_with_old_rule_schema_version_failure(self):
+        with self.swap(
+            feconf, 'CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION', 2):
+            with self.assertRaisesRegexp(
+                Exception,
+                'Current platform parameter rule schema version is v2, '
```

Comment:
This exception may easily break if the schema version changes. Suggest interpolating it to ensure it stays up-to-date.

Reply:
I think it's fine here because I wrapped the test in the `self.swap` scope that mocks the `feconf.CURRENT_PLATFORM_PARAMETER_RULE_SCHEMA_VERSION = 2`.